### PR TITLE
feat(analyzer): disable analyzers

### PR DESCRIPTION
**Pull request by knqyf263**
This PR makes it possible to disable analyzers.

### CODE REVIEW

**Comment by masahiro331** [comment]
IMO.
When sharing cache in server mode, The disable-analyzer can have an unexpected negative cache .
For example, John  use trivy command with removed-packages option, after that Fred use trivy command.
At this point Fred receives the result of running removed-pkgs.

File: artifact/remote/git.go [hunk 1]
```diff
@@ -38,7 +40,10 @@ func NewArtifact(rawurl string, c cache.ArtifactCache) (artifact.Artifact, func(
 		_ = os.RemoveAll(tmpDir)
 	}
 
-	return local.NewArtifact(tmpDir, c), cleanup, nil
+	// JAR/WAR/EAR doesn't need to be analyzed in git repositories.
+	disabled = append(disabled, analyzer.TypeJar)
```
Comment (masahiro331) [inline artifact/remote/git.go:44]: Just my preference:
I think it is better to implement this process in Trivy.

Comment (knqyf263) [reply artifact/remote/git.go:44]: Some analyzers should not be enabled regardless of callers. I think it is better to hard-code and disable them here.

**Comment by knqyf263** [comment]
@masahiro331 Nice catch. The change would be large, so I'll work on it in a dedicated PR.

File: artifact/remote/git.go (hunk 1 above)
Comment (masahiro331) [reply artifact/remote/git.go:44]: I got it.

**Comment by masahiro331** [comment]
LGTM! :shipit:

**Review by masahiro331 — APPROVED** [review]

**Pull request merged**